app/src/main/java/de/danoeh/antennapod/dialog/EpisodesApplyActionFragment.java

```diff
@@ -118,62 +164,93 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                 } else {
+                    hideSpeedDial();
+                }
+            }
+
+            @Override
+            public void onScroll(AbsListView view, int firstVisibleItem, int visibleItemCount, int totalItemCount) {}
+        });
 
         for(FeedItem episode : episodes) {
             titles.add(episode.getTitle());
         }
 
         mAdapter = new ArrayAdapter<>(getActivity(),
-                android.R.layout.simple_list_item_multiple_choice, titles);
+                R.layout.simple_list_item_multiple_choice_on_start, titles);
         mListView.setAdapter(mAdapter);
-        checkAll();
 
-        int lastVisibleDiv = 0;
-        btnAddToQueue = view.findViewById(R.id.btnAddToQueue);
-        if((actions & ACTION_QUEUE) != 0) {
-            btnAddToQueue.setOnClickListener(v -> queueChecked());
-            lastVisibleDiv = R.id.divider1;
-        } else {
-            btnAddToQueue.setVisibility(View.GONE);
-            view.findViewById(R.id.divider1).setVisibility(View.GONE);
-        }
-        btnMarkAsPlayed = view.findViewById(R.id.btnMarkAsPlayed);
-        if((actions & ACTION_MARK_PLAYED) != 0) {
-            btnMarkAsPlayed.setOnClickListener(v -> markedCheckedPlayed());
-            lastVisibleDiv = R.id.divider2;
-        } else {
-            btnMarkAsPlayed.setVisibility(View.GONE);
-            view.findViewById(R.id.divider2).setVisibility(View.GONE);
-        }
-        btnMarkAsUnplayed = view.findViewById(R.id.btnMarkAsUnplayed);
-        if((actions & ACTION_MARK_UNPLAYED) != 0) {
-            btnMarkAsUnplayed.setOnClickListener(v -> markedCheckedUnplayed());
-            lastVisibleDiv = R.id.divider3;
-        } else {
-            btnMarkAsUnplayed.setVisibility(View.GONE);
-            view.findViewById(R.id.divider3).setVisibility(View.GONE);
-        }
-        btnDownload = view.findViewById(R.id.btnDownload);
-        if((actions & ACTION_DOWNLOAD) != 0) {
-            btnDownload.setOnClickListener(v -> downloadChecked());
-            lastVisibleDiv = R.id.divider4;
-        } else {
-            btnDownload.setVisibility(View.GONE);
-            view.findViewById(R.id.divider4).setVisibility(View.GONE);
-        }
-        btnDelete = view.findViewById(R.id.btnDelete);
-        if((actions & ACTION_REMOVE) != 0) {
-            btnDelete.setOnClickListener(v -> deleteChecked());
-        } else {
-            btnDelete.setVisibility(View.GONE);
-            if(lastVisibleDiv > 0) {
-                view.findViewById(lastVisibleDiv).setVisibility(View.GONE);
+        saveActionBarTitle(); // needed when we dynamically change the title based on selection
+
+        // Init action UI (via a FAB Speed Dial)
+        mSpeedDialView = view.findViewById(R.id.fabSD);
+        mSpeedDialView.inflate(R.menu.episodes_apply_action_speeddial);
+
+        // show only specified actions, and bind speed dial UIs to the actual logic
+        for (ActionBinding binding : actionBindings) {
```

Comment:
Nice!

Reply:
Thanks. I actually was of two minds when I refactored it - The conciseness comes with a cost of a few bindings objects created every time a new fragment instance created - it's probably not significant enough to worry about it.